src/Renderer/NavigatorRenderer.cpp

```diff
@@ -0,0 +1,378 @@
+/*
+Copyright_License {
+
+  XCSoar Glide Computer - http://www.xcsoar.org/
+  Copyright (C) 2000-2021 The XCSoar Project
+  A detailed list of copyright holders can be found in the file "AUTHORS".
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU General Public License
+  as published by the Free Software Foundation; either version 2
+  of the License, or (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+}
+*/
+
+#include "NavigatorRenderer.hpp"
+#include "Formatter/Units.hpp"
+#include "InfoBoxes/InfoBoxLayout.hpp"
+#include "Interface.hpp"
+#include "Language/Language.hpp"
+#include "Look/FontDescription.hpp"
+#include "Look/Look.hpp"
+#include "Math/Angle.hpp"
+#include "NextArrowRenderer.hpp"
+#include "Renderer/TextRenderer.hpp"
+#include "Task/TaskBehaviour.hpp"
+#include "Units/System.hpp"
+#include "Waypoint/Waypoint.hpp"
+#include "WaypointIconRenderer.hpp"
+#include "WaypointRenderer.hpp"
+#include "time/RoughTime.hpp"
+#include "ui/canvas/Canvas.hpp"
+#include "Screen/Layout.hpp"
+#include "UIGlobals.hpp"
+#include "Look/TaskLook.hpp"
+#include "Look/WaypointLook.hpp"
+#include "Look/NavigatorLook.hpp"
+#include "Look/IconLook.hpp"
+#include "Look/MapLook.hpp"
+#include "Look/TaskLook.hpp"
+#include "Math/Constants.hpp"
+#include "ProgressBarRenderer.hpp"
+#include "Task/ProtectedTaskManager.hpp"
+#include "Engine/Task/TaskManager.hpp"
+#include "Engine/Task/Ordered/OrderedTask.hpp"
+#include "Engine/Task/Ordered/Points/OrderedTaskPoint.hpp"
+#include "Components.hpp"
+#include "Engine/Task/Unordered/AlternateList.hpp"
+#include "Formatter/LocalTimeFormatter.hpp"
+#include "Formatter/TimeFormatter.hpp"
+#include "ui/canvas/Font.hpp"
+#include "ui/dim/Rect.hpp"
+
+
+// standard
+#include <iomanip>
+#include <iostream>
+#include <string>
+
+
+void
+NavigatorRenderer::DrawFrame(Canvas &canvas, const PixelRect &rc, const NavigatorLook &look) {
+  const auto top_left = rc.GetTopLeft();
+  const int rc_height = rc.GetHeight();
+  const int rc_width = rc.GetWidth();
+  
+  BulkPixelPoint polyline_frame[11];
```

Comment:
No magic numbers please - `const BulkPixelPoint polyline_frame[] = { ...`

Reply:
Oh, all those variables ... that's ugly ....

Reply:
I propose 2 versions: one with shortened variable names (my preferred), and one with additional comments.

1st version:
I propose to shorten the name of the variables. it's not necessary to use too long variable name in this scope. It also lets the code more readable and allows not to add comments. Actually, I wanted to use variables in this case to use them within the next points. that could be also very helpful if minimal modifications have to be done in the polygone_frame. (change of the size, of the width, ...)

2nd version:
I propose a variant without variable but I didn't find it more readable. I have to add comments to understand how it is coded whereas in the previous version with variables, all is in the code directly.

```
// --------------- 1st version with variables --------------- //
  // the divisions below are not simplified deliberately to understand frame construction
  const PixelPoint pt0 = {top_left.x + rc_height * 5 / 20, top_left.y};
  const PixelPoint pt1 = {top_left.x + rc_width - rc_height * 4 / 20, pt0.y};
  const PixelPoint pt2 = {top_left.x + rc_width - rc_height * 2 / 20, 
                          top_left.y + rc_height * 1 / 20};
  const PixelPoint pt3 = {top_left.x + rc_width, 
                          top_left.y + rc_height * 10 / 20};
  const PixelPoint pt4 = {pt2.x, top_left.y + rc_height * 19 / 20};
  const PixelPoint pt5 = {pt1.x, top_left.y + rc_height};
  const PixelPoint pt6 = {pt0.x, pt5.y};
  const PixelPoint pt7 = {top_left.x + rc_height * 2 / 20, pt4.y};
  const PixelPoint pt8 = {top_left.x, pt3.y};
  const PixelPoint pt9 = {pt7.x, pt2.y};
  
  const BulkPixelPoint polygone_frame[] = {pt0, pt1, pt2, pt3, pt4, pt5, pt6, pt7, pt8, pt9};


// --------------- 2nd version with comments --------------- //
  const BulkPixelPoint polygone_frame1[] = {
  {top_left.x + rc_height * 5 / 20, top_left.y},                            // pt0 = ..., ...
  {top_left.x + rc_width - rc_height * 4 / 20, top_left.y},                 // pt1 = ..., pt0.y
  {top_left.x + rc_width - rc_height * 2 / 20,                              // pt2 = ..., ...
   top_left.y + rc_height * 1 / 20},
  {top_left.x + rc_width, top_left.y + rc_height * 10 / 20},                // pt3 = ..., ...
  {top_left.x + rc_width - rc_height * 2 / 20,                              // pt4 = pt2.x, ...
   top_left.y + rc_height * 19 / 20},                                       
  {top_left.x + rc_width - rc_height * 4 / 20, top_left.y + rc_height},     // pt5 = pt1.X, ... 
  {top_left.x + rc_height * 5 / 20, top_left.y + rc_height},                // pt6 = pt0.x, pt5.y
  {top_left.x + rc_height * 2 / 20, top_left.y + rc_height * 19 / 20},      // pt7 = ..., pt4.y
  {top_left.x, top_left.y + rc_height * 10 / 20},                           // pt8 = ..., pt3.y
  {top_left.x + rc_height * 2 / 20, top_left.y + rc_height * 1 / 20}};      // pt9 = pt7.x, pt2.y   
```

Reply:
do you have preference for 1st or 2nd version ? 
or other idea ?

Reply:
OK, first version kept.
see [comment](https://github.com/XCSoar/XCSoar/pull/1064#discussion_r1535879215)